xarray/tests/test_combine.py

```diff
@@ -661,7 +661,18 @@ def test_merge_one_dim_concat_another(self):
                  Dataset({'foo': ('x', [2, 3])})],
                 [Dataset({'bar': ('x', [10, 20])}),
                  Dataset({'bar': ('x', [30, 40])})]]
-        actual = auto_combine(objs, concat_dim=[None, 'x'])
+        actual = auto_combine(objs, concat_dim=[None, 'x'], compat='equals')
+        assert_identical(expected, actual)
+
+    def test_internal_odering(self):
```

Comment:
Does this test fail with the old logic?

Reply:
Yes, it gives a `MergeError`. 

Though maybe I need to dig more, because I thought it would pass with `compat='no_conflicts'`, and fail with `compat='equals'`...